File: packages/terse/src/tiers/rules/fillers.ts
// Filler words/phrases removed only at sentence-start position.
// Mid-sentence removal is too risky without POS tagging (handled in NLP tier).
//
// Inclusion criteria: safe to strip when the word adds rhetorical color but does not change
// the truth value or uncertainty of the claim that follows.
//
// Intentionally excluded (qualify the claim — carry semantic weight):
//   theoretically, technically, realistically, ideally, presumably, supposedly, arguably(*),
//   apparently — these signal caveat, uncertainty, or ideal-vs-actual distinctions.
//   (*) arguably is borderline but often signals genuine dispute; left out to be safe.
//
// "of course" is a two-word phrase but safe at sentence start: "Of course, this works." → "this works."

const SENTENCE_START_FILLERS =
  /(?<=(?:^|[.!?])\s{0,3})(basically|essentially|simply|obviously|clearly|literally|honestly|actually|frankly|truthfully|personally|genuinely|admittedly|undoubtedly|certainly|naturally|importantly|notably|interestingly|surprisingly|unsurprisingly|thankfully|fortunately|unfortunately|luckily|of course),?\s+/gim;

export function removeFillers(text: string): string {
  return text.replace(SENTENCE_START_FILLERS, "");
}

File: crates/terse/src/tiers/rules/fillers.rs
use once_cell::sync::Lazy;
use regex::Regex;

// Sentence-start filler words.
// Strategy: capture the preceding sentence-end + optional whitespace as group 1,
// then replace with just group 1 (preserving the sentence boundary).
// (?im): case-insensitive + multiline so ^ matches line starts.
static SENTENCE_START_FILLERS: Lazy<Regex> = Lazy::new(|| {
    Regex::new(
        r"(?im)((?:^|[.!?])\s{0,3})(basically|essentially|simply|obviously|clearly|literally|honestly|actually|frankly|truthfully|personally|genuinely|admittedly|undoubtedly|certainly|naturally|importantly|notably|interestingly|surprisingly|unsurprisingly|thankfully|fortunately|unfortunately|luckily|of course),?\s+"
    )
    .expect("invalid filler regex")
});

pub fn remove_fillers(text: &str) -> String {
    // Replace with group 1 (the sentence-start boundary) to preserve structure
    SENTENCE_START_FILLERS
        .replace_all(text, "$1")
        .into_owned()
}
